util.go

```diff
@@ -10,26 +11,58 @@ import (
 
 // WaitFor polls a predicate function, once per second, up to a timeout limit.
 // It usually does this to wait for a resource to transition to a certain state.
+// To handle situations when the predicate response might hang indefinitely,
+// the predicate will be retried every "retry" seconds.
 // Resource packages will wrap this in a more convenient function that's
 // specific to a certain resource, but it can also be useful on its own.
-func WaitFor(timeout int, predicate func() (bool, error)) error {
-	start := time.Now().Second()
-	for {
-		// Force a 1s sleep
-		time.Sleep(1 * time.Second)
+func WaitFor(timeout, retry int, predicate func() (bool, error)) error {
```

Comment:
At the moment, this doesn't work on a timeout (e.g. 
```go
WaitFor(1, 0, func()(bool, error) error {
  time.Sleep(10000000 * time.Second)
  return true, nil
}
```

I think the changes mentioned below should fix it though.